tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1874,7 +1874,12 @@ IrEmitterUnnested::BuildKernelThunkFromBufferSlices(
     if (alloc.IsPreallocatedTempBuffer()) {
       if (!temp_buffer.has_value()) {
         temp_buffer = &alloc;
-      } else {
+      } else if (inst->parent()
+                     ->parent()
+                     ->config()
+                     .debug_options()
+                     .xla_multiheap_size_constraint_per_heap() == -1) {
+        // Check temp buffer numbers only when the multiheap mode is off.
```

Comment:
I simply skip the check when the multiheap mode is on. It works fine (through NV's CI and to the best of my knowledge) as I don't see any codes replying on the single temp buffer assumption for kernel code generation.

The only place related to this may be that the last kernel argument is assumed a temp buffer. (But again I don't see codes using this temp buffer.)

Simply allow multiple temp buffers works but please feel free to suggest better ways to deal with this piece of code.